src/wallet/rpcwallet.cpp

```diff
@@ -2454,67 +2456,72 @@ Value zc_raw_receive(const json_spirit::Array& params, bool fHelp)
 
     LOCK(cs_main);
```

Comment:
I agree. This lock is really only needed inside of `WitnessBucketCommitment` as far as I can tell.